### PR TITLE
Various integration test improvements

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

**Comment by bluca** [comment]
This needs a change in the CentOS tests to switch to ncat and stress-ng - FYI @mrc0mmand

File: .github/workflows/mkosi.yml [hunk 1]
```diff
@@ -92,7 +92,7 @@ jobs:
 
     steps:
     - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332
-    - uses: systemd/mkosi@70aa901697f12182ccaa24e2325867d275479b55
+    - uses: systemd/mkosi@50946d3fdf7381fe1a1c004a0ed6c668b2f7d38f
```
Comment (keszybz) [inline .github/workflows/mkosi.yml]: Hmm, now I'm wishing for something like `tools/update-distro-hash`…

File: mkosi.images/system/mkosi.conf [hunk 2]
```diff
@@ -15,8 +15,10 @@ ExtraTrees=
 
 PostInstallationScripts=mkosi.sanitizers.chroot
 
+Environment=
+        SYSTEMD_REPART_OVERRIDE_FSTYPE_ROOT=%F
```
Comment (keszybz) [inline mkosi.images/system/mkosi.conf]: Probably an ignorant question, but why is this needed if the default filesystem is the default?

**Review by keszybz — APPROVED** [review]
Looks good in principle.

File: mkosi.images/system/mkosi.conf (hunk 2 above)
Comment (daandemeyer) [reply mkosi.images/system/mkosi.conf]: Because the default filesystem only applies when mkosi generates its own repart partition definition files. This bit is to make it apply with the custom configuration we ship in mkosi.repart/
